File: python/bowling_game/bowling_game.py
class BowlingGame():
    def __init__(self) -> None:
        self.rolls = []
        pass

    def roll(self, pins: int) -> None:
        self.rolls.append(pins)
        pass

    def score(self) -> int:
        score = 0
        frame_index = 0
        for frame in range(10):
            if self.is_strike(frame_index):
                score += 10 + self.strike_bonus(frame_index)
                frame_index += 1
            elif self.is_spare(frame_index):
                score += 10 + self.spare_bonus(frame_index)
                frame_index += 2
            else:
                score += self.rolls[frame_index] + self.rolls[frame_index + 1]
                frame_index += 2
        return score

    def is_strike(self, frame_index: int) -> bool:
        return self.rolls[frame_index] == 10

    def is_spare(self, frame_index: int) -> bool:
        return self.rolls[frame_index] + self.rolls[frame_index + 1] == 10

    def strike_bonus(self, frame_index: int) -> int:
        return self.rolls[frame_index + 1] + self.rolls[frame_index + 2]

    def spare_bonus(self, frame_index: int) -> int:
        return self.rolls[frame_index + 2]

File: java_maven/BowlingGameKata/src/main/java/steelpipe/BowlingGame.java
package steelpipe;

public class BowlingGame {

    private int[] rolls = new int[21];
    private int currentRoll = 0;

    public void roll(int pins) {
        rolls[currentRoll++] = pins;
    }

    public int score() {
        int score = 0;
        int rollIndex = 0;

        for (int frame = 0; frame < 10; frame++) {
            if (rolls[rollIndex] == 10) { // ストライクの場合
                score += 10 + rolls[rollIndex + 1] + rolls[rollIndex + 2];
                rollIndex++;
            } else if (rolls[rollIndex] + rolls[rollIndex + 1] == 10) { // スペアの場合
                score += 10 + rolls[rollIndex + 2];
                rollIndex += 2;
            } else { // その他の場合
                score += rolls[rollIndex] + rolls[rollIndex + 1];
                rollIndex += 2;
            }
        }

        return score;
    }
}
